### PR TITLE
Added smarty plugins page to designers guide

### PR DESCRIPTION
**Pull request by wolv-dev**

### CODE REVIEW

File: source/designers-guide/smarty-plugins/index.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+http://myshop.de/examples
+``` 
+ 
+As an additional parameter a title can be specified for the Smarty plugin, which is then displayed with the old SEO links in the URL.
+ 
+ 
+Example: *plugin call with a set parameter*
+``` 
+{* Syntax *}
+{"shopware.php?sViewport=cat&sCategory=32"|rewrite:"examples"}
```
Comment (wolv-dev) [inline source/designers-guide/smarty-plugins/index.md]: Hab das Rewrite Plugin entfernt.

File: source/designers-guide/index.html [hunk 2]
```diff
@@ -28,11 +28,13 @@
     <li><a href="{{ site.url }}/designers-guide/snippets/">Snippet management</a></li>
     <li><a href="{{ site.url }}/designers-guide/external-resources/">Embedding external resources</a></li>
     <li><a href="{{ site.url }}/designers-guide/responsive-images/">Working with responsive images</a></li>
+    <li><a href="{{ site.url }}/designers-guide/smarty-plugins/">Smarty plugins</a></li>
     <li><a href="{{ site.url }}/designers-guide/best-practice-theme-development/">Using Grunt for theme development</a></li>
     <li><a href="{{ site.url }}/designers-guide/modify-jquery-plugins">Modify jQuery plugins</a></li>
     <li><a href="{{ site.url }}/designers-guide/managing-third-party-dependencies-with-bower/">Managing third-party dependencies with Bower</a></li>
     <li><a href="{{ site.url }}/designers-guide/managing-third-party-dependencies-with-npm/">Managing third-party dependencies with NPM</a></li>
     <li><a href="{{ site.url }}/designers-guide/legacy-template-development-in-shopware-5/">Legacy template development in Shopware 5</a></li>
+    <li><a href="{{ site.url }}/designers-guide/smarty-plugins/">Smarty plugins</a></li>
```
Comment (Phil23) [inline source/designers-guide/index.html]: The entry is duplicated. Please just add one link in the correct section.

Please add also the link in the same order in `source/designers-guide/general-resources.html`.

File: source/designers-guide/smarty-plugins/index.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
+---
+layout: default
+title: Smarty Plugins
+github_link: designers-guide/smarty-plugins/index.md
+indexed: true
```
Comment (Phil23) [inline source/designers-guide/smarty-plugins/index.md]: Please add the info for the menu structure to also add a menu link.

For example:
```
group: Frontend Guides
subgroup: General Resources
menu_title: Smarty Plugins
menu_order: 55
```

File: source/designers-guide/smarty-plugins/index.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+<div class="alert alert-info">
+	Note: Shopware, by default, formats prices in the frontend as follows:
+	{$sArticle.price|currency:use_symbol:right}.
+	If you wish to have certain areas of your shop in a different format, this plugin will allow you to change the format.
+</div>
+
+## Config plugin ##
+
+The config plugin is grants you the ability to access and read the various features in the backend.
```
Comment (Phil23) [inline source/designers-guide/smarty-plugins/index.md]: "The config plugin grants"

File: source/designers-guide/smarty-plugins/index.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+Without the plugin, the call looks like this:
+
+Example: *normal style sheet call*
+```
+<link rel="stylesheet" media="screen, projection" href="http://www.meinshop.de/shopware/templates/my_template/frontend/_resources/styles/my_styles.css" />
+```
+
+
+By using the link plugin, you no longer need to specify an exact path, merely to go through the selected template.
```
Comment (Phil23) [inline source/designers-guide/smarty-plugins/index.md]: "By using the link plugin you only have to define the path relative to your own theme directory."

File: source/designers-guide/smarty-plugins/index.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+The following types can be defined for the plugin:
+
+- ISO - ISO format for date formatting
+- PHP - PHP's date() function for date formatting
+
+ 
+## Action plugin ##
+
+The action plugin is one of Shopware 4's new features. The plugin uses widgets to embed templates. Widgets are self-contained parts of the frontend, such as the new shopping worlds.
```
Comment (Phil23) [inline source/designers-guide/smarty-plugins/index.md]: Please do not reference to Shopware 4 :)

Please rewrite this section.

**Comment by mitelg** [comment]
hey @wolv-dev any updates here? :blush:

File: source/designers-guide/smarty-plugins/index.md (hunk 3 above)
Comment (wolv-dev) [reply source/designers-guide/smarty-plugins/index.md]: added

**Comment by wolv-dev** [comment]
@mitelg Sorry, missed the notifications. Now updated :)

File: source/designers-guide/smarty-plugins/index.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+</a>
+``` 
+ 
+The syntax of the url plugin is as follows:
+ 
+ 
+Example: *Syntax url plugin*
+``` 
+{* Syntax *}
+{url controller='[CONTROLLERNAME]' action='[ACTIONNAME]' [MORE PARAMETERS='PARAMETERVALUE']}
```
Comment (mitelg) [inline source/designers-guide/smarty-plugins/index.md]: the `module` parameter is missing here. it defaults to `frontend` but `widgets` is also a use case

File: source/designers-guide/smarty-plugins/index.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+The plugin also automatically builds up the SEO links if the corresponding plugin is installed.
+
+## Date plugin ##
+
+The date plugin is used to format date and time information. To this end, you have the ability to use a variety of formats. The syntax of the date plugin is as follows:
+
+Example: *date plugin syntax*
+``` 
+{* Syntax *}
+{[WERT]|date:[FORMAT]:[TYP]}
```
Comment (mitelg) [inline source/designers-guide/smarty-plugins/index.md]: please use english here :wink:

File: source/designers-guide/smarty-plugins/index.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+- RSS - date for RSS feeds e.g., "Thu, 04 Nov 2010 13:55:52 +0100"
+- COOKIE - date for cookies e.g., "Thursday, 04-Nov-10 1:55:52 p.m. Europe / Berlin"
+- W3C - date for HTML or HTTP W3C e.g., "2010-11-04T13:55:52+01:00"
+
+The following types can be defined for the plugin:
+
+- ISO - ISO format for date formatting
+- PHP - PHP's date() function for date formatting
+
+<<<<<<< HEAD
```
Comment (mitelg) [inline source/designers-guide/smarty-plugins/index.md]: I think this could be deleted

File: source/designers-guide/smarty-plugins/index.md [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+<<<<<<< HEAD
+=======
+
+## Rewrite plugin ##
```
Comment (mitelg) [inline source/designers-guide/smarty-plugins/index.md]: the rewrite plugin does not exist anymore. I think an old commit is still here :blush:

**Comment by mitelg** [comment]
hey @wolv-dev 
thanks for the update. I found some things, which should be fixed. after that I'm looking forward to merge your PR :+1:

**Comment by mitelg** [comment]
now we got it :smiley: :+1: 
thanks for your contribution @wolv-dev :tada:

**Pull request merged**